tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -1190,6 +1190,8 @@ uint64_t SparseGlobalOrderReader::compute_var_size_offsets(
           break;
       }
 
+      total_cells--;
```

Comment:
I think you are on the right track here but I took some time to review the whole function this morning and found a couple issues. I will send you an updated version in slack that also tried to clarify a few comments.